source-code/core/config/src/schema.ts

```diff
@@ -0,0 +1,79 @@
+import { Static, Type } from "@sinclair/typebox"
+
+/**
+ * The options of a plugin.
+ *
+ * The options are defined by a user via the `PluginSettings`.
+ *
+ * ```diff
+ * └── config: InlangConfig
+ *   └── settings: InlangConfig["settings"]
+ *       └── plugins: PluginSettings
+ * +         ├── options: PluginOptions
+ *           └── ...
+ * ```
+ */
+export type PluginOptions = Static<typeof PluginOptions>
+export const PluginOptions = Type.Record(
+	Type.String(),
+	Type.Union([Type.String(), Type.Array(Type.String()), Type.Record(Type.String(), Type.String())]),
+)
+
+/**
+ * The options of a lint rule.
+ *
+ * The options are defined by a user via the `LintRuleSettings`.
+ *
+ * ```diff
+ * └── config: InlangConfig
+ *   └── settings: InlangConfig["settings"]
+ * +      └── lintRule: LintRuleSettings
+ *           ├── options: PluginOptions
+ *           └── ...other properties of LintRuleSettings
+ * ```
+ */
+export type LintRuleSettings = Static<typeof LintRuleSettings>
+export const LintRuleSettings = Type.Object({
+	options: Type.Optional(PluginOptions),
+	level: Type.Optional(
+		Type.Union([Type.Literal("off"), Type.Literal("warning"), Type.Literal("error")]),
+	),
+})
+
+/**
+ * The settings of a plugin.
+ *
+ * ```diff
+ * └── config: InlangConfig
+ * +  └── settings: InlangConfig["settings"]
+ *       └── plugins: PluginSettings
+ *           ├── options: PluginOptions
+ *           └── ...
+ * ```
+ */
+export type PluginSettings = Static<typeof PluginSettings>
+export const PluginSettings = Type.Object({
+	options: Type.Optional(PluginOptions),
+})
+
+export type InlangConfig = Static<typeof InlangConfig>
+export const InlangConfig = Type.Object({
```

Comment:
we should probably invest some time into making error messages more human readable

Reply:
![image](https://github.com/inlang/inlang/assets/21335119/0d47f788-a99f-482c-87fa-f3614eb22639)

Reply:
Another one:
![image](https://github.com/inlang/inlang/assets/21335119/6d595a34-a48f-4793-a674-386f03cb2b12)